File: backend/src/main/scala/services/battle/microservices/actors/objects/player/BattleAppearanceKeys.scala
package services.battle.microservices.actors.objects.player

final case class BattleAvatarKey private (value: String) extends AnyVal
final case class BattleSkinKey private (value: String) extends AnyVal

object BattleAvatarKey {
  def fromWire(value: String): Option[BattleAvatarKey] =
    Option(value).map(_.trim).filter(_.nonEmpty).map(new BattleAvatarKey(_))
}

object BattleSkinKey {
  def fromWire(value: String): Option[BattleSkinKey] =
    Option(value).map(_.trim).filter(_.nonEmpty).map(new BattleSkinKey(_))
}

File: frontend/src/objects/battle/microservices/actors/objects/player/BattleAppearanceKeys.ts
export type BattleAvatarKey = string;
export type BattleSkinKey = string;

export function battleAvatarKeyFromWire(value: string): BattleAvatarKey | null {
  const trimmed = value.trim();
  return trimmed.length > 0 ? trimmed : null;
}

export function battleSkinKeyFromWire(value: string): BattleSkinKey | null {
  const trimmed = value.trim();
  return trimmed.length > 0 ? trimmed : null;
}

